skt/runner.py

```diff
@@ -274,7 +274,7 @@ def cancel_pending_jobs(self):
             if ret:
                 logging.info('Failed to cancel the remaining recipe sets!')
 
-        for job_id in self.job_to_recipe_set_map:
+        for job_id in set(self.job_to_recipe_set_map):
```

Comment:
You forgot to push the test associated with this bugfix :stuck_out_tongue:

Reply:
I'd like to push those tests en-masse. It's kindof tedious to open a pull request for every patch.

Reply:
Probably, but also is the best way to get your changes merged fast.

Reply:
I've added the test on top.

Reply:
Wow, thanks!